fastlane/Fastfile

```diff
@@ -0,0 +1,20 @@
+# This file contains the fastlane.tools configuration
+# You can find the documentation on https://docs.fastlane.tools
+#
+# For a list of all available actions, check out
+#
+#     https://docs.fastlane.tools/actions
+#
+
+default_platform(:ios)
+
+platform :ios do
+  desc "Run all unit tests for ImagineEngine (best engine)"
+  lane :tests do
+    run_tests(scheme: "ImagineEngine-iOS")
```

Comment:
Won't this mean that you'll run the tests for all platforms on iOS? 🤔

Reply:
The generated command is

```
set -o pipefail && env NSUnbufferedIO=YES xcodebuild -scheme ImagineEngine-iOS -project ./ImagineEngine.xcodeproj -destination 'platform=iOS Simulator,id=2A0FBBBD-6835-4D69-88FA-B1CE91BF4675' -derivedDataPath '/Users/felixkrause/Library/Developer/Xcode/DerivedData/ImagineEngine-bmlfowclistvrdedznkikhhmwtqo' build test | tee '/Users/felixkrause/Library/Logs/scan/ImagineEngine-ImagineEngine-iOS.log' | xcpretty  --report html --output '/Users/felixkrause/Developer/hacking/ImagineEngine/fastlane/test_output/report.html' --report junit --output '/Users/felixkrause/Developer/hacking/ImagineEngine/fastlane/test_output/report.junit' --report junit --output '/var/folders/px/bz2kts9n69g8crgv4jpjh6b40000gn/T/junit_report20180108-30088-8dlwit'
```

simplified:
```
xcodebuild -scheme ImagineEngine-iOS -project ./ImagineEngine.xcodeproj -destination 'platform=iOS Simulator,id=2A0FBBBD-6835-4D69-88FA-B1CE91BF4675' build test | xcpretty
```

Reply:
So it defines a specific iOS simulator:
```
-destination 'platform=iOS Simulator,id=2A0FBBBD-6835-4D69-88FA-B1CE91BF4675'
```

Reply:
Yeah, but on the next line you are running the macOS tests, but on line `11` we specify `platform :ios`, seems conflicting? Or am I missing something?

Reply:
Ahh, I see what you're saying. The platform `:ios` means it's not Android. Is that confusing? I'll remove it for your Fastfile, as UnrealEngine probably won't support Android. But I'm asking, should we in general never generate the `platform :ios` piece?

Reply:
Ah, I see. I think it makes sense for fastlane in general since I would assume it's mainly used for iOS or Android apps. But in this context it was a bit confusing, since we're building an iOS/Mac/tvOS project. Maybe one day this project will be as big as UnrealEngine 😜